resource_customizations/cluster.x-k8s.io/MachineHealthCheck/health.lua

```diff
@@ -0,0 +1,10 @@
+hs = {}
+
+hs.status = "Healthy"
+hs.message = ""
```

Comment:
It's a bit unusual for health checks to be healthy by default. Could it instead be "Progressing" and then move to "Healthy" based on some condition?

Reply:
Hmm I dont know, honestly I know very little about this resource (and argo custom health checks in general)
You can't really see if its doing something, ie progressing. Thats why I left it with just 2 states.
Either expectedMachines == currentHealthy, or it's not :)

Reply:
If you want, I just set it to default progressing,
if  expectedMachines == currentHealthy, healthy
if expectedMachines != currentHealthy, degraded

but its really just the same thing as this

Reply:
Well, I think it probably needs a nil check for `obj.status`. So if someone applies a MachineHealthCheck, but the controller for that CRD isn't running, we will show "Progressing" instead of "Degraded" (or, in this case, having a nil pointer error).

Reply:
Fixed